File: day21/1.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {

  int a = strtol(argv[1], NULL, 10);
  int b = strtol(argv[2], NULL, 10);

  long a_score = 0;
  long b_score = 0;
  int turn = 0;
  long dice = 1;
  long t = 0;

  while (a_score < 1000 && b_score < 1000) {
    t += 3;
    long roll = 3 * dice + 3;
    dice += 3;
    if (!turn) {
      a = (a + roll) % 10;
      if(!a){
        a = 10;
      }
      a_score += a;
      turn++;
    }
    else{
      b = (b + roll)%10;
      if(!b){
        b = 10;
      }
      b_score += b;
      turn--;
    }
  }
  long min;
  if (a_score < b_score){
    min = a_score;
  }
  else{
    min = b_score;
  }
  printf("%ld \n",min*t);
  return 0;
}

File: day21/1.py
from sys import argv

a = int(argv[1])
b = int(argv[2])

a_score = 0
b_score = 0
turn = 0
dice = 1
t = 0

while a_score < 1000 and b_score < 1000:
    t += 3
    roll = 3*dice + 3
    dice += 3
    if turn == 0:
        a = (a+roll) % 10
        if a == 0:
            a = 10
        a_score += a
        turn += 1
    else:
        b = (b + roll) % 10
        if b == 0:
            b = 10
        b_score += b
        turn -= 1
    if turn == 0:
        print(f"2 score: {a_score}")
    else:
        print(f"1 score: {b_score}")
print("-----end------")
print(a_score, b_score, t)
print(f"part one solution: {min(a_score,b_score)*t}")
